test/NCI.OCPL.Api.CTSListingPages.Tests/Tests/Services/ESLabelLookupQueryServiceTest.Get.cs

```diff
@@ -0,0 +1,206 @@
+
+            // Setup the mocked Options
+            IOptions<ListingPageAPIOptions> clientOptions = GetMockOptions();
+
+            Mock<AbstractLogger<ESLabelLookupQueryService>> _mockLogger = new Mock<AbstractLogger<ESLabelLookupQueryService>>();
+
+            ESLabelLookupQueryService queryService = new ESLabelLookupQueryService(client, clientOptions, _mockLogger.Object);
+
+            APIInternalException ex = await Assert.ThrowsAsync<APIInternalException>(() => queryService.Get("chicken"));
+
+            // Verify the correct error message is thrown the correct number of times.
+            _mockLogger.Verify(x => x.Log(Microsoft.Extensions.Logging.LogLevel.Error, It.IsAny<Exception>(), "Invalid response when searching for pretty URL name or identifier 'chicken'."), Times.Once);
+        }
+
+        public static IEnumerable<object[]> Get_Success_Scenarios = new[]
+        {
+            new object[] { new Get_NoResults() },
+            new object[] { new Get_SingleResult() },
+            new object[] { new Get_MultipleResults() }
+        };
+
+        /// <summary>
+        /// Test to verify handling when a get request results in no errors (no results, single result, multiple results).
+        /// </summary>
+        [Theory, MemberData(nameof(Get_Success_Scenarios))]
+        public async void Get_TestSuccess(Get_BaseScenario data)
+        {
+            ElasticsearchInterceptingConnection conn = new ElasticsearchInterceptingConnection();
+            conn.RegisterRequestHandlerForType<Nest.SearchResponse<LabelInformation>>((req, res) =>
+            {
+                res.Stream = TestingTools.GetStringAsStream(data.MockESResponse);
+                res.StatusCode = 200;
+            });
+
+            // The URI does not matter, an InMemoryConnection never requests from the server.
+            var pool = new SingleNodeConnectionPool(new Uri("http://localhost:9200"));
+
+            var connectionSettings = new ConnectionSettings(pool, conn);
+            IElasticClient client = new ElasticClient(connectionSettings);
+
+            // Setup the mocked Options
+            IOptions<ListingPageAPIOptions> clientOptions = GetMockOptions();
+
+            Mock<AbstractLogger<ESLabelLookupQueryService>> _mockLogger = new Mock<AbstractLogger<ESLabelLookupQueryService>>();
+
+            ESLabelLookupQueryService query = new ESLabelLookupQueryService(client, clientOptions, _mockLogger.Object);
+
+            LabelInformation result = await query.Get("treatment");
+
+            // Verify that the logger logs a warning the expected number of times, with the expected error message.
+            // For no results and one result, the logger logs zero warnings with no error message.
+            // For multiple results, the logger logs one warning with the expected error message.
+            _mockLogger.Verify(x => x.Log(Microsoft.Extensions.Logging.LogLevel.Warning, It.IsAny<Exception>(), data.ErrorMessage), data.TimesCalled);
```

Comment:
Similar to above, the `Verify` call becomes something similar to this:

```csharp
_mockLogger.Verify(
    x => x.Log(
        It.Is<Microsoft.Extensions.Logging.LogLevel>(l => l == Microsoft.Extensions.Logging.LogLevel.Warning),
        It.IsAny<EventId>(),
        It.Is<It.IsAnyType>((v, t) => v.ToString() == data.ErrorMessage),
        It.IsAny<Exception>(),
        It.Is<Func<It.IsAnyType, Exception, string>>((v, t) => true)
    ),
    Moq.Times.Exactly(data.ExpectedLoggingCalls)
);
```
Note that I've changed the "Number of times called" test to retrieve an integer from the data object, rather than a `Moq.Times` struct.  The data objects should be lightweight with (preferably) no dependencies.